CONTRIBUTING.md

```diff
@@ -1,256 +1,263 @@
 
-## Local development
+See [doc/README.md](doc/README.md).
 
-### 1: Preparations
+# Setting up for Helm chart development
 
-Before anything else, install [git](https://www.git-scm.com/), clone the
-repository, and enter the repository directory.
 
-```
-git clone https://github.com/jupyterhub/zero-to-jupyterhub-k8s
-cd zero-to-jupyterhub-k8s
-```
 
-For local development, you will additional tools and we present you with two
-options. Either you, a) start and work from a Virtual Machine that is
-automatically prepared for development, or b) install the tools yourself and
-work without a Virtual Machine.
+## 1: Prerequisites
 
-#### a) Use a prepared Virtual Machine (VM)
+This needs to be installed:
 
-1. Install VirtualBox by [downloading and running an
-   installer](https://www.virtualbox.org/wiki/Downloads).
+- [docker](https://docs.docker.com/get-docker/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (also [setup tab completion](https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion))
+- [helm](https://helm.sh/docs/intro/install/) (also [setup tab completion](https://helm.sh/docs/helm/helm_completion/))
+- Python 3.7+ ([Anaconda.com](https://www.anaconda.com/distribution/),
+  [Python.org](https://www.python.org/downloads/)) and dependencies:
+  ```shell
+  pip install -r dev-requirements.txt
+  pip install -r doc/doc-requirements.txt
+  ```
 
-1. Install Vagrant by [downloading and running an
-   installer](https://www.vagrantup.com/downloads.html).
+You can check if you have it all like this:
 
-1. Start and automatically setup a VM.
+```shell
+docker --version
+kubectl version --client
+helm version
+pytest --version
+chartpress --version
+```
 
-   Use the `vagrant up` command to start the VM for the first time, or `vagrant
-   resume` to resume a suspended VM.
+## 2: Setup a Kubernetes cluster
 
-   ```shell
-   vagrant up
-   # vagrant resume
-   ```
+We need a Kubernetes cluster to work against.
 
-1. Enter the VM and the `~/zero-to-jupyterhub-k8s` folder.
+### Linux only: Kubernetes setup with k3s
+With [k3s](https://github.com/rancher/k3s) we can _quickly_ create a Kubernetes
+cluster, and we _don't have to transfer docker images_ built on our computer to
+make them available in the Kubernetes cluster.
 
-   The `~/zero-to-jupyterhub-k8s` folder in the VM will be the exact same folder
-   on your actual machine. Change either and you influence both.
+__Install__
 
-   ```shell
-   vagrant ssh
-   cd zero-to-jupyterhub-k8s
-   ```
+```shell
+# Installs a ~50 MB k3s binary, setups and starts a systemctl service called
+# k3s which is also enabled to run on startup, provides a k3s-uninstall.sh
+# script, disables not needed functionality. You will be asked for sudo rights
+curl -sfL https://get.k3s.io | sh -s - \
+    --write-kubeconfig-mode=644 \
+    --disable metrics-server \
+    --disable traefik \
+    --disable local-storage \
+    --disable-network-policy \
```

Comment:
Oh... I forgot that now we may want to have the step to install the CNI in this docs as well, because --flannel-backend=none is only suitable along with setting up calico.

Reply:
Hmmmm... I wonder if i think it make more sense to have a section on running netpol tests. They add quite a bit of complexity to the setup, needing --flannel-backend=none during k3s install + calico installed afterwards with a patched configuration. And, i've not tested it on k3d either and bet it can be different in that situation.

WDYT, okay to make a section about netpol tests that we reference instead of forcing everybody to install the tools needed for them?

Reply:
I'll make another pass on the netpol instructions, perhaps clearing away canal steps that didn't pan out in Travis CI which reduce some complexity and avoids raising questions about canal etc.

I'm strongly biased towards:
- only supporting what we are confident we can maintain
- minimal amount of helper scripts